std/experimental/allocator/building_blocks/aligned_block_list.d

```diff
@@ -0,0 +1,629 @@
+        to the nearest multiple of the `alignment` to quickly find the corresponding
+        `AlignedBlockNode`.
+        */
+        bool deallocate(void[] b);
+
+        /**
+        Returns `Ternary.yes` if the buffer belongs to the parent allocator and
+        `Ternary.no` otherwise.
+        */
+        static if (hasMember!(ParentAllocator, "owns"))
+        Ternary owns(void[] b);
+    }
+    else
+    {
+        import std.math : isPowerOf2;
+        static assert (isPowerOf2(alignment));
+        mixin AlignedBlockListImpl!true;
+    }
+}
+
+///
+@system unittest
+{
+    import std.experimental.allocator.building_blocks.region : SharedRegion;
+    import std.experimental.allocator.building_blocks.ascending_page_allocator : SharedAscendingPageAllocator;
+    import core.thread : ThreadGroup;
+
+    enum numThreads = 8;
+    enum size = 2048;
+    enum maxIter = 10;
+
+    /*
+    In this example we use 'SharedAlignedBlockList' together with 'SharedRegion',
+    in order to create a fast, thread-safe allocator.
+    */
+    alias SuperAllocator = SharedAlignedBlockList!(
+            SharedRegion!(NullAllocator, 1),
+            SharedAscendingPageAllocator,
+            4096);
+
+    SuperAllocator a;
+    // The 'SuperAllocator' will draw memory from a 'SharedAscendingPageAllocator'
+    a.parent = SharedAscendingPageAllocator(4096 * 1024);
+
+    // Launch 'numThreads', each performing allocations
+    void fun()
+    {
+        foreach(i; 0 .. maxIter)
+        {
+            void[] b = a.allocate(size);
+            assert(b.length == size);
+        }
+    }
+
+    auto tg = new ThreadGroup;
+    foreach (i; 0 .. numThreads)
+    {
+        tg.create(&fun);
+    }
+    tg.joinAll();
+}
+
+version (unittest)
+{
+    static void testrw(void[] b)
+    {
+        ubyte* buf = cast(ubyte*) b.ptr;
+        size_t len = (b.length).roundUpToMultipleOf(4096);
+        for (int i = 0; i < len; i += 4096)
+        {
+            buf[i] =  (cast(ubyte)i % 256);
+            assert(buf[i] == (cast(ubyte)i % 256));
+        }
+    }
+}
+
+@system unittest
+{
+    import std.experimental.allocator.building_blocks.region;
+    import std.experimental.allocator.building_blocks.ascending_page_allocator;
+    import std.random;
+    import std.algorithm.sorting : sort;
+    import core.thread : ThreadGroup;
+    import core.internal.spinlock : SpinLock;
+
+    enum pageSize = 4096;
+    enum numThreads = 10;
+    enum maxIter = 20;
+    enum totalAllocs = maxIter * numThreads;
+    size_t count = 0;
+    SpinLock lock = SpinLock(SpinLock.Contention.brief);
+
+    alias SuperAllocator = SharedAlignedBlockList!(
+            SharedRegion!(NullAllocator, 1),
+            SharedAscendingPageAllocator,
+            1 << 16);
+    void[][totalAllocs] buf;
+
+    SuperAllocator a;
+    a.parent = SharedAscendingPageAllocator(4096 * 1024);
+
+    void fun()
+    {
+        auto rnd = Random();
```

Comment:
No. Use a fixed seed and a fixed engine, everything else might lead to random coverage changes.